### DISCUSSION TITLE
Workers vs Child Process and single file executable

### DISCUSSION THREAD

**Discussion by rgillan** [oven-sh/bun#14053] [Q&A] [answered]
Currently we have implemented a "cluster" of Bun.serve/websocket servers using spawned child processes.  To package these up we have been separately generating single file executables, one for the primary and the other for the children.
Now that the single-file executable compile process will allow workers to be included, it would be an obvious choice to move from spawned children to worker threads so we can ship a single executable (not two).  The API docs still show the Worker API as experimental.
So wondering what the best approach here is (currently) for a production environment where we need single file executables?  Should we retain the current approach (child process) and ship 2 binaries, or is the experimental tag on the worker approach simply due to it not being battle tested and we can ship a single executable with both primary and worker with little risk?
Any thoughts/guidance very much appreciated.

**Comment by Jarred-Sumner** [comment, marked as answer]
There are yet-to-be-fixed reliability and memory usage issues that can happen when a Worker is terminated frequently. If you don't ever terminate the Worker though, it should work fine.

**Comment by rgillan** [comment]
@Jarred-Sumner thanks as always for the prompt response.  The workers should never be terminated in our case but we might err on sticking with the current child spawn approach for now (as we have battle/load tested that config).  Will keep an eye on the docs as shipping 1 binary is more desirable.  Cheers
